### DISCUSSION TITLE
How do I get an alert if my job doesn’t finish successfully

### DISCUSSION THREAD

**Discussion by yuhan** [dagster-io/dagster#7159] [Q&A] [answered]
Does Dagster provide native support for this?

**Comment by yuhan** [comment, marked as answer]
Dagster has built-in support for job-level alerting called `run_status_sensor`. See examples here: https://docs.dagster.io/concepts/partitions-schedules-sensors/sensors#run-status-sensors

**Reply by markfickett** [reply]
A sensor can do this, but there are also [builtin alerts](https://docs.dagster.io/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui) which are available in the UI under Deployments > Alerts, which can handle a few common cases: job failure, job success, or job running for more than N hours.

![image](https://github.com/dagster-io/dagster/assets/307950/969371b2-ffa1-4b1d-97e5-669ce58a610d)

**Reply by solorez** [reply]
> A sensor can do this, but there are also [builtin alerts](https://docs.dagster.io/dagster-plus/managing-deployments/alerts/managing-alerts-in-ui) which are available in the UI under Deployments > Alerts, which can handle a few common cases: job failure, job success, or job running for more than N hours.
> 
> ![image](https://private-user-images.githubusercontent.com/307950/336491540-969371b2-ffa1-4b1d-97e5-669ce58a610d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3ODk1NjcsIm5iZiI6MTczMzc4OTI2NywicGF0aCI6Ii8zMDc5NTAvMzM2NDkxNTQwLTk2OTM3MWIyLWZmYTEtNGIxZC05N2U1LTY2OWNlNThhNjEwZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQwMDA3NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iZWE3Y2YxNTAwNGIxYzg4YzQ4YzRlMmQ4M2FlNTlkN2JkOTQ0MzE3ZTRhMDc1ZDIxYzlmZTllMTkyYjEwMDhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CsA9xNNwYJ42_25H7Jhi1heNnDaq8u9-RItPiZWYrdU)

it seems this function is offered in dagster+.
